backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt

```diff
@@ -289,4 +289,56 @@ class ShorkInterpreterControllerV1IT : AbstractControllerTest() {
 
         assertEquals(HttpStatusCode.BadRequest, result.status)
     }
+
+    @Test
+    fun `test join lobby with valid playerName`() = runTest {
+        client.post("/api/v1/lobby") {
+            contentType(ContentType.Application.Json)
+            setBody("{\"playerName\":\"playerA\"}")
+        }
+        val result =
+            client.post("/api/v1/lobby/0/join") {
+                contentType(ContentType.Application.Json)
+                setBody("{\"playerName\":\"playerB\"}")
+            }
+        assertEquals(HttpStatusCode.OK, result.status)
+    }
+
+    @Test
+    fun `test join lobby with duplicate (invalid) playerName`() = runTest {
+        client.post("/api/v1/lobby") {
+            contentType(ContentType.Application.Json)
+            setBody("{\"playerName\":\"playerA\"}")
+        }
+        val result =
+            client.post("/api/v1/lobby/0/join") {
+                contentType(ContentType.Application.Json)
+                setBody("{\"playerName\":\"playerA\"}")
+            }
+        assertEquals(HttpStatusCode.Conflict, result.status)
+    }
+
+    @Test
+    fun `test join lobby with invalid json`() = runTest {
+        client.post("/api/v1/lobby") {
+            contentType(ContentType.Application.Json)
+            setBody("{\"playerName\":\"playerA\"}")
+        }
+        val result =
+            client.post("/api/v1/lobby/0/join") {
+                contentType(ContentType.Application.Json)
+                setBody("{ invalid, : json :3")
+            }
+        assertEquals(HttpStatusCode.BadRequest, result.status)
+    }
+
+    @Test
+    fun `test join nonexistent lobby`() = runTest {
+        val result =
+            client.post("/api/v1/lobby/0/join") {
+                contentType(ContentType.Application.Json)
+                setBody("{\"playerName\":\"playerA\"}")
+            }
+        assertEquals(HttpStatusCode.Conflict, result.status)
+    }
```

Comment:
This test is good, but I feel like a 404 would be better here. Opinions? Should we adjust the spec?

Reply:
404 if the lobby does not exist, `conflict` if it's disallowed due to the player already being in the lobby is fitting imo

Reply:
yes, my thoughts.

I will adjust the docs accordingly then.